shell.nix

```diff
@@ -0,0 +1,53 @@
+{ pkgs ? import <nixpkgs> {} }:
+	let
+	overrides = (builtins.fromTOML (builtins.readFile ./rust-toolchain.toml));
+	libPath = with pkgs; lib.makeLibraryPath [
+		# load external libraries that you need in your rust project here
+	];
+in
+pkgs.mkShell rec {
+	# Kanidm dependencies
+	buildInputs = with pkgs; [
+		pkg-config
+		
+		git
+		rustup
+
+		clang
+		llvmPackages.bintools
+		
+		openssl
+		systemd
+		linux-pam
+	];
```

Comment:
```suggestion
    ] ++ pkgs.lib.optionals (pkgs.stdenv.isLinux) [
		systemd
		linux-pam
	];
```

Only valid on Linux

Reply:
I don't think we can build the pam module on mac? I'll fix it up but running `cargo build` at the root would fail on macos because it relies on linux-pam

Reply:
It’ll be ignored, there’s a bunch of work we did to make this OK.

Reply:
It’ll be ignored, there’s a bunch of work we did to make this OK.

Reply:
But also don’t try and install Linux PAM support libs on macOS 😳

Reply:
that is what this suggestion is for. Implemented it since it just makes sense in any case